src/rules/noReturnInFinallyRule.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-return-in-finally",
+        description: "Disallows return statements in finally blocks.",
+        descriptionDetails: "",
+        rationale: "Return statements inside finally blocks have confusing semantics.",
```

Comment:
Would it be acceptable to link to eslint.org (seems a bit weird)? Or maybe: "Using control-flow statements such as return, continue, break and throws inside finally blocks overrides any control-flow statements in the same try ... catch scope. This is considered confusing and unexpected behavior."